### PR TITLE
[RESTEASY-1992] Prepare tests for extended support for multivalue @*Param

### PR DESCRIPTION
**Pull request by marekkopecky**
Add aditionall tests for extended support for multivalue @*Param

* Rename (and improve) original `MultiValuedParamTest` to `MultiValuedParamDateProxyTest`
   * This test uses java.util.Date class as multivalue param
   * This test uses proxy (String parameter is defined in proxy, server uses collections in end-point), proxy is not related with this feature
* `MultiValuedParamDateNoProxyTest`
     * Same as `MultiValuedParamDateProxyTest`, but proxy is not used at all in this test
* `MultiValuedParamCustomClassTest`
    * This test uses custom deployment class
* `MultiValuedParamCdiTest`
    * Server end-point is defined as a CDI bean with @RequestScoped annotation

Jira: https://issues.jboss.org/browse/RESTEASY-1992
3.6.1 PR: https://github.com/resteasy/Resteasy/pull/1660
3.6 PR: https://github.com/resteasy/Resteasy/pull/1658

### CODE REVIEW

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCdiTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import org.junit.runner.RunWith;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.core.Response;
+
+/**
+ * @tpSubChapter Parameters
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails Test of CDI integration for RESTEasy extended support for multivalue @*Param (RESTEASY-1566 + RESTEASY-1746)
+ * @tpSince RESTEasy 3.6.1.Final
```
Comment (ronsigal) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCdiTest.java]: 4.0.0

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCustomClassTest.java [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import javax.ws.rs.core.Response;
+
+/**
+ * @tpSubChapter Parameters
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails Test for RESTEasy extended support for multivalue @*Param (RESTEASY-1566 + RESTEASY-1746)
+ *                    org.jboss.resteasy.test.resource.param.resource.MultiValuedParamPersonWithConverter class is used
+ * @tpSince RESTEasy 3.6.1.Final
```
Comment (ronsigal) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCustomClassTest.java]: 4.0.0

**Comment by ronsigal** [comment]
@marekkopecky , very nice.

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCustomClassTest.java [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+     */
+    @Test
+    public void testCookieParam() throws Exception {
+        ResteasyClient client = new ResteasyClientBuilder().build();
+        try {
+            Response response;
+            String cookieExpectedResponse = person1.toString();
+
+            response = client.target(generateBaseUrl() + "/cookieParam/customConversion_list")
+                    .request().cookie("person",name1).get();
```
[the remark this answers is not in the document]
Comment (marekkopecky) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCustomClassTest.java]: My mistake, of course more persons can be used here. Cookies doesn't allow to use ',', see the spec (https://tools.ietf.org/html/rfc6265), so we need to use '-' as a separator. I updated converters like [this](https://github.com/resteasy/Resteasy/pull/1658/files#diff-0c3cf7de5c965575e5a7315f449889faR15)

**Comment by marekkopecky** [comment]
@ronsigal PR updated, thank you for the review!

**Pull request merged**